src/test/UnbondingLock.t.sol

```diff
@@ -0,0 +1,101 @@
+contract UnbondingLockTest is L2ArbitrumMessenger, DSTest {
+    ICheatCodes public constant CHEATS = ICheatCodes(HEVM_ADDRESS);
+
+    address public constant L1_MIGRATOR_ADDRESS =
+        0x21146B872D3A95d2cF9afeD03eE5a783DaE9A89A;
+    L2Migrator public constant L2_MIGRATOR =
+        L2Migrator(payable(0x148D5b6B4df9530c7C76A810bd1Cdf69EC4c2085));
+    IBondingManagerOverride public constant BONDING_MANAGER =
+        IBondingManagerOverride(0x35Bcf3c30594191d53231E4FF333E8A770453e40);
+
+    address public constant L1_DELEGATE =
+        0xa20416801aC2eACf2372e825B4a90ef52490c2Bb;
+    address public constant DELEGATOR_WITH_NON_NULL_DELEGATE =
+        0x0070EdA17A3656D6568eC4C94FeF34A396D20613;
+    address public constant DELEGATOR_WITH_NULL_DELEGATE =
+        0x00ac9C5193660b1F69092Fb4B0f011521B67627f;
+    IRoundsManager public constant ROUND_MANAGER =
+        IRoundsManager(0xdd6f56DcC28D3F5f27084381fE8Df634985cc39f);
+
+    uint256 public constant TEST_BLOCK_NUMBER = 9999999999;
+
+    function testFinalizeMigrateUnbondingLocksDelegatorWithNullDelegate()
+        public
+    {
+        CHEATS.roll(TEST_BLOCK_NUMBER);
+        ROUND_MANAGER.initializeRound();
+        CHEATS.startPrank(applyL1ToL2Alias(L1_MIGRATOR_ADDRESS));
+        address delegateAddress = _migrateBondingLock(
+            DELEGATOR_WITH_NULL_DELEGATE,
+            address(0)
+        );
+        CHEATS.stopPrank();
+        assertTrue(delegateAddress == address(0));
+    }
+
+    function testDelegateTransferWithDelegatorWithNullDelegate() public {
+        CHEATS.roll(TEST_BLOCK_NUMBER);
+        ROUND_MANAGER.initializeRound();
+        CHEATS.startPrank(applyL1ToL2Alias(L1_MIGRATOR_ADDRESS));
+        address l2DelegateAlias = applyL1ToL2Alias(L1_DELEGATE);
+        address delegateAddress = _migrateBondingLock(
+            DELEGATOR_WITH_NULL_DELEGATE,
+            l2DelegateAlias
+        );
+        CHEATS.stopPrank();
+        assertTrue(delegateAddress == l2DelegateAlias);
+    }
+
+    function testDelegateTransferWithDelegatorWithNonNullDelegate() public {
+        CHEATS.roll(TEST_BLOCK_NUMBER);
+        ROUND_MANAGER.initializeRound();
+        CHEATS.startPrank(applyL1ToL2Alias(L1_MIGRATOR_ADDRESS));
+        address l2DelegateAlias = applyL1ToL2Alias(L1_DELEGATE);
+        address delegateAddress = _migrateBondingLock(
+            DELEGATOR_WITH_NON_NULL_DELEGATE,
+            l2DelegateAlias
+        );
+        CHEATS.stopPrank();
+        assertTrue(delegateAddress == l2DelegateAlias);
+    }
+
+    function _migrateBondingLock(address _delegator, address _delegate)
+        private
+        returns (address)
+    {
+        uint256[] memory _unbondingLockIds = new uint256[](0);
+        IMigrator.MigrateUnbondingLocksParams
+            memory migrateDelegatorParams = IMigrator
+                .MigrateUnbondingLocksParams({
+                    l1Addr: _delegator,
+                    l2Addr: applyL1ToL2Alias(_delegator),
```

Comment:
Applying alias is not recommended for EOAs unless the plan was to receive the locks to an address different than the original address.

On Arbitrum, when an L1 contract calls a function on L2 via the inbox, its address is masked. hence, `msg.sender` for L2 call becomes (L1Address + mask) and because of that we use the aliased address.

In case of EOA, The addresses remain same across the chains.
Moreover, It is unlikely that the user will have the private keys to the masked address and thus anything sent over to the masked address will become unusable.

Reply:
good point, thanks!